docs/sphinx/developer/arch/supported/aarch64.rst

```diff
@@ -142,7 +145,19 @@ There are several useful variables that execution behaviours have access to:
+Scalable Vector Extension
+''''''''''''''''''''''''''
+SimEng supports the Arm SVE extension and thus the use of ``Z`` vector registers. ``Z`` registers are an extension of the Arm NEON ``V`` vector registers whereby the ``V`` register variant occupies the lower 16 bytes of the ``Z`` registers total 256 bytes. Under the Arm SVE extension, the implemented logic for writing to a ``V`` register is to zero-out the upper 240 bytes of the associated ``Z`` register (e.g. ``z1`` and ``v1``) and treat its lower 16 bytes as the ``V`` register. SimEng will automatically apply this logic when the execution of an instruction contains a ``V`` register as a destination location.
+
+Scalable Matrix Extension
+''''''''''''''''''''''''''
+Also supported is the Arm SME extension and thus the use of ``ZA`` sub-tile registers. The implementation of the ``ZA`` register is to tread each horizontal row the same as a vector (or ``Z``) register. Therefore, if a source operand is a sub-tile of ``ZA`` and contains 16 rows, then there will be 16 corresponding entries in the ``operands`` vector. Likewise, if a destination operand is ``ZA`` or a sub-tile of ``ZA`` then the ``results`` vector will require the corresponding number of horizontal rows. 
+
+One quirk of SME instructions comes with sub-tile slices; individual rows or column within a sub-tile. Regardless of whether a whole sub-tile or a slice is used as a source operand, all rows associated with said tile will be added to the ``operands`` vector. There are two reasons for this. First, the index value pointing to the relevant slice cannot be evaluated before instruction execution, thus, all sub-tile rows need to be provided. Second, if the source slice is a vertical slice (or a column of the sub-tile) then an element from each row is needed to construct the correct output.
+
+Furthermore, a similar situation is present when a sub-tile slice is a destination operand. The ``results`` vector will expect a ``registerValue`` entry for each row of the targetted sub-tile, again due to the same two reasons listed previously. But, when a sub-tile slice is a destination operand, **all** associated rows of the sub-tile will be also added to the ``operands`` vector. Again, this is down to two key, similar reasons. First, when a destination is a sub-tile slice, we only want to update that row or column. Therefore, in order to not zero-out the other slices within the sub-tile we will need access to their current values. Secondly, if the destination is a vertical slice (or sub-tile column) then only one element per row should be updated; the rest should remain unchanged.
```

Comment:
I should re-order my points in the docs to make this clearer - but we do not know which slice will be updated until execute due to the index which determines the slice being a `wn` register. 

Therefore, we need to add the whole tile to the destination operands as it could be any of them that is updated, or all of them if it is a vertical slice (column).

However, if no value is provided to `results` in execute, it will default to a RegisterValue of 0. And for vertical slice updates we only update a single element per row (or update an entire column) whilst maintaining the values of all other elements.

Reply:
I have added and updated this section. Please check